docs/enhancements/third-party-cni-support.md

```diff
@@ -0,0 +1,269 @@
+
+**Documentation Links:**
+Each CNI option displays a link to vendor-specific documentation for post-install configuration and manifest examples.
+
+#### Kubernetes API Changes
+
+Update `AgentClusterInstall` CR to support CNI selection through a new `networkType` field in the `spec.networking` section. Supported values include `OVNKubernetes`, `OpenShiftSDN`, `CiscoACI`, `Cilium`, `Calico`, and `None`.
+
+The validation webhook will enforce CNI constraints, ensuring `userManagedNetworking: true` is set for third-party CNIs and validating against the support matrix. OpenShiftSDN availability will be validated based on OCP version compatibility.
+
+#### CNI Support Matrix Storage
+
+The CNI support matrix will be stored as structured data in `data/cni_support_matrix.json`, containing:
+- CNI names and display names
+- Supported OCP versions per CNI
+- Supported platforms per CNI/OCP version combination
+- Version-specific constraints
+- Documentation URLs for each CNI vendor
+
+This structured storage enables easier updates without code changes and allows future dynamic fetching of support data from external services.
+
+### Risks and Mitigations
+
+**CNI Support Matrix Maintenance**
+The CNI certification program updates regularly. The support matrix embedded in Assisted Installer may become outdated.
+
+*Mitigation:* Implement dynamic support matrix fetching (similar to OCP release versions). Add CI validation to detect drift from published certification data. Consider maintaining matrix in a separate service for easier updates.
+
+**Testing Coverage**
+Assisted Installer CI is optimized for OVN-K. Third-party CNI testing may be insufficient without vendor participation.
+
+*Mitigation:* Partner with CNI vendors (Cisco, Isovalent, Tigera) to establish joint testing programs. Mark feature as TechPreview initially. Require CNI vendors to maintain E2E test suites against Assisted Installer. Progress to GA after sufficient validation.
+
+**Support Complexity**
+Separating Assisted Installer issues from CNI operator issues may be challenging for support teams.
+
+*Mitigation:* Implement clear logging boundaries. Develop diagnostic tools to validate CNI installation status. Create support runbooks for triaging CNI-related issues. Establish escalation paths to CNI vendor support teams. These diagnostic and support tools will be developed in Phase 3 of the implementation.
+
+**Backward Compatibility**
+Introducing CNI selection may affect existing installation workflows.
+
+*Mitigation:* Default to OVN-K for all existing behavior. Treat third-party CNI as opt-in feature requiring explicit selection. Ensure clusters without `network_type` specified continue using OVN-K. Document that CNI selection is Day 1 only (no Day 2 migration).
+
+## Design Details
+
+### Phased Implementation
+
+**Phase 1: "No CNI" Mode (TechPreview)**
+- Introduce feature flag `ENABLE_NO_CNI_MODE` to allow cluster installation without OVN-K
```

Comment:
Will this feature flag get removed in later phases or will its default change in later phases?